signinapp/model.py

```diff
@@ -219,18 +225,24 @@ def make(username: str, name: str, password: str, role: Role, approved=False, su
         if "phone_number" in kwargs:
             kwargs["phone_number"] = normalize_phone_number_for_storage(
                 kwargs["phone_number"])
-        return User(username=username,
+        user = User(username=username,
                     name=name,
                     password=generate_password_hash(password),
                     code=mk_hash(name),
                     role_id=role.id,
                     subteam_id=subteam.id if subteam else 0,
                     approved=approved, **kwargs)
+        db.session.add(user)
+        db.session.flush()
+        return user
 
     @staticmethod
     def make_guardian(name: str, phone_number: str, email: str, contact_order: int = 1):
         role = Role.from_name("guardian_limited")
+        # Generate a username that *should* be unique.
+        username = f"{name}_{normalize_phone_number_for_storage(phone_number)}"
```

Comment:
We'll need some way to change the username when they create a full guardian account

Reply:
Yeah, we'll have to handle this as part of the special guardian registration page. (Match user on name and phone number instead of username)